src/util/mod.rs

```diff
@@ -683,6 +683,23 @@ pub(crate) unsafe fn copy_unchecked(src: &[u8], dst: &mut [u8]) {
     };
 }
 
+/// Unsafely transmutes the given `src` into a type `Dst`.
+///
+/// # Safety
+///
+/// The value `src` must be a bit-valid instance of `Dst`.
+#[inline(always)]
+pub(crate) const unsafe fn transmute_unchecked<Src, Dst>(src: Src) -> Dst {
+    static_assert!(Src, Dst => core::mem::size_of::<Src>() == core::mem::size_of::<Dst>());
```

Comment:
The safety precondition is sufficient, as it will allow us to eventually support padding-extension transmutations without updating the SAFETY comment of every other call-site. (We'll still need to update the SAFETY doc comment of `transmute_from`, of course.)

There's no point in making this a macro that desugars to `mem::transmute`. The whole point of this is to have something usable in generic contexts, since `mem::transmute` isn't.